src/System.Text.Primitives/System/Text/Parsing/Unsigned.cs

```diff
@@ -0,0 +1,301 @@
+﻿
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// NOTE: This file is generated via a T4 template. Please do not edit this file directly. Any changes should be made
+// in Unsigned.tt.
+
+namespace System.Text
+{
+    public static partial class PrimitiveParser
+    {
+        public static bool TryParseByte(ReadOnlySpan<byte> text, out byte value, out int bytesConsumed, EncodingData encoding = default(EncodingData), TextFormat format = default(TextFormat))
+        {
+            if (format.IsHexadecimal)
+            {
+                if (encoding.IsInvariantUtf8)
+                {
+                    return InvariantUtf8.Hex.TryParseByte(text, out value, out bytesConsumed);
+                }
+                else if (encoding.IsInvariantUtf16)
+                {
+                    ReadOnlySpan<char> utf16Text = text.Cast<byte, char>();
+                    int charsConsumed;
+                    bool result = InvariantUtf16.Hex.TryParseByte(utf16Text, out value, out charsConsumed);
+                    bytesConsumed = charsConsumed * sizeof(char);
+                    return result;
+                }
+                else
+                {
+                    // TODO: EncodingData.InvariantUtf8 and EncodingData.InvariantUtf16 don't work for parsing right now
+                    // because their parsing trees are undefined. A custom EncodingData with a manually-created parsing
+                    // tree must be used instead. Default to InvariantUtf8 parsing for now, but after autogeneration of
+                    // parsing trees is implemented, this should return false.
+                    return InvariantUtf8.Hex.TryParseByte(text, out value, out bytesConsumed);
```

Comment:
I am not sure I understand. If the encoding is invariant, your code should either work or throw NYI exceptions. If the encoding is not invariant, for now we should probably throw NotSupportedException, and then think if we want to support it later and how.